paddle/fluid/eager/utils.cc

```diff
@@ -288,7 +288,9 @@ void EagerUtils::CheckAndRetainGrad(
 
 paddle::experimental::Tensor EagerUtils::SyncToPtenTensors(
     const egr::EagerTensor& tensor) {
-  const_cast<EagerTensor*>(&tensor)->SyncToTensor();
+  if (!tensor.initialized()) {
```

Comment:
Don't need this after [PR39376](https://github.com/PaddlePaddle/Paddle/pull/39376)